crypto/digestible/derive/test/tests/behavior.rs

```diff
@@ -108,32 +126,114 @@ fn bar1() {
         e: u32::max_value(),
         f: Foo { a: 5, b: 6, c: 7 },
     };
-    let mut hasher = Tester::new();
-    arg.digest(&mut hasher);
-
-    let expected: Vec<Vec<u8>> = vec![
-        b"Bar".to_vec(),
-        b"".to_vec(),
-        b"d".to_vec(),
-        b"Blob".to_vec(),
-        b"".to_vec(),
-        b"0".to_vec(),
-        5usize.to_le_bytes().to_vec(),
-        b"Koala".to_vec(),
-        b"e".to_vec(),
-        vec![255u8, 255u8, 255u8, 255u8],
-        b"f".to_vec(),
-        b"Foo".to_vec(),
-        b"".to_vec(),
-        b"a".to_vec(),
-        vec![5u8, 0u8],
-        b"b".to_vec(),
-        vec![6u8, 0u8],
-        b"c".to_vec(),
-        vec![7u8, 0u8],
-    ];
-
-    assert_eq!(hasher.args, expected);
+    let expected_ast = ASTNode::from(ASTAggregate {
+        context: b"bar1",
+        name: b"Bar".to_vec(),
+        elems: vec![
+            ASTNode::from(ASTAggregate {
+                context: b"d",
+                name: b"Blob".to_vec(),
+                elems: vec![ASTNode::from(ASTPrimitive {
+                    context: b"0",
+                    type_name: b"bytes",
+                    data: b"Koala".to_vec(),
+                })],
+                is_completed: true,
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"e",
+                type_name: b"uint",
+                data: vec![255u8; 4],
+            }),
+            ASTNode::from(ASTAggregate {
+                context: b"f",
+                name: b"Foo".to_vec(),
+                elems: vec![
+                    ASTNode::from(ASTPrimitive {
+                        context: b"a",
+                        type_name: b"uint",
+                        data: vec![5u8, 0u8],
+                    }),
+                    ASTNode::from(ASTPrimitive {
+                        context: b"b",
+                        type_name: b"uint",
+                        data: vec![6u8, 0u8],
+                    }),
+                    ASTNode::from(ASTPrimitive {
+                        context: b"c",
+                        type_name: b"uint",
+                        data: vec![7u8, 0u8],
+                    }),
+                ],
+                is_completed: true,
+            }),
+        ],
+        is_completed: true,
+    });
+    digestible_test_case_ast("bar1", &arg, expected_ast);
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"bar1"),
+        [
+            237, 231, 204, 138, 55, 249, 219, 0, 154, 213, 236, 77, 123, 104, 185, 68, 165, 117,
+            179, 15, 85, 65, 13, 134, 163, 16, 206, 60, 249, 184, 194, 81
+        ]
+    );
+}
+
+#[test]
+fn bar2() {
```

Comment:
As a general comment, I'm finding all the `foos` and `bars` to be unhelpful identifiers. For example, I think a Bar is a "normal, C-style struct" and a TBar is a "normal C-syle struct containing a transparent field". That could all be clarified with comments, but the tests might be a lot more self-documenting if the type names were meaningful, i.e. Bar -> PlainStruct, TBar -> StructWithTransField, etc.

Reply:
yeah this is a good criticism, thanks i will fix